alacritty/src/display/content.rs

```diff
@@ -257,6 +259,7 @@ impl RenderableCell {
 
         RenderableCell {
             zerowidth: cell.zerowidth().map(|zerowidth| zerowidth.to_vec()),
+            hyperlink: cell.hyperlink().cloned(),
```

Comment:
Haven't looked at the full PR yet, but do we really need to clone this for every single cell part of the hyperlink? If it's only about rendering wouldn't it suffice to make this into a bool?

Reply:
I also don't want this field be a part of `RenderableCell` since it doesn't affect rendering.
It's here because in `Display::draw` we convert the grid to `RenderableCell`, then drop the grid early to free the lock, before traversing cells here: https://github.com/alacritty/alacritty/blob/770128c16d1baea1d8f1b528937f38226ca58c02/alacritty/src/display/mod.rs#L530

If it's not in `RenderableCell`, we should move the highlight checking logic to here, when holding the lock:
https://github.com/alacritty/alacritty/blob/770128c16d1baea1d8f1b528937f38226ca58c02/alacritty/src/display/mod.rs#L491

Reply:
That only needs the bounds though, right? We don't need to clone the arc?

Reply:
> That only needs the bounds though

Yes but we cannot get bounds from `Hyperlink` without traversing the grid.
Traversing grid and comparing them with highlighted one when holding the term lock seems quite wrong.

Another idea is to traverse the grid in `highlighted_at`, so it only occurs when mouse hovering a hyperlink. And save *ALL* bounds instead of the hovered one to `HintMatch`. Then we'll have bounds when rendering.

Reply:
I'm not sure what you mean. This is just stored here to render the underline, right? Why would we need to store the URL's **text** to render that? I don't see any reason why we'd need that information.

Storing IDs to compare to the hovered one to distinguish between hovered or not is one thing, but that doesn't require any text storage.

Reply:
> Storing IDs to compare to the hovered one

I got it. But id can also be `Arc<str>` in some cases. You still cannot get rid of string or Arc cloning. I don't think replacing `Hyperlink` with `Id` here really changes anything.

Reply:
Well dragging around an Arc is definitely different from dragging around an integer. It just seems a bit wasteful to clone everything for every single cell, especially if there's a lot of URIs present.